pkg/console/controllers/route/controller.go

```diff
@@ -159,6 +186,48 @@ func (c *RouteSyncController) SyncRoute(operatorConfig *operatorsv1.Console) (co
 	return rt, rtIsNew, "", rtErr
 }
 
+// Custom route sync needs to:
+// 1. validate if the reference for secret with serving-cert is defined in operator config(in case a non-openshift cluster domain is used)
+// 2. sync the secret with the serving-cert, from the openshift-config,into the openshift-console namespace
+// 3. create the custom console route
+// 4. check custom route health
+// 5. validate the route
+func (c *RouteSyncController) SyncCustomRoute(operatorConfig *operatorsv1.Console) (consoleRoute *routev1.Route, isNew bool, reason string, err error) {
+	if configErr := routesub.ValidateCustomRoute(operatorConfig, c.routeClient); configErr != nil {
+		return nil, true, "InvalidCustomRouteConfig", configErr
+	}
+
+	if scscErr := c.SyncCustomServigCertSecret(operatorConfig); scscErr != nil {
+		return nil, false, "FailedSyncServingCertSecret", scscErr
+	}
+
+	cr, crIsNew, crErr := routesub.GetOrCreate(c.routeClient, routesub.CustomRoute(operatorConfig))
+	if crErr != nil {
+		return nil, false, "FailedCreateCustomRoute", crErr
+	}
+	c.CheckRouteHealth(operatorConfig, cr)
+
+	if _, changed := routesub.Validate(cr); changed {
+		return nil, true, "InvalidCustomRoute", customerrors.NewSyncError("custom route is invalid, correcting route state")
+	}
+
+	return cr, crIsNew, "", crErr
+}
+
+func (c *RouteSyncController) SyncCustomServigCertSecret(operatorConfig *operatorsv1.Console) error {
```

Comment:
why are you syncing this? Also, I'd advise to use the name "RouteCert" instead of serving-cert because the latter is more commonly used for the certificate the pod uses for serving, rather than the certificate the route uses.

Reply:
> why are you syncing this?

we need to since the secret that contains the custom serving-cert from `openshift-config`, where the user/admin needs to create it manually, into the `openshift-console`, so we can mount it into the `console` pod

Reply:
But you don't want to mount it in the pod, you want to have that inserted into the route